### ISSUE TITLE
RDATEs are not being evaluated

### ISSUE THREAD

**Issue by robertsuditu** [ical-org/ical.net#107]
The RecurringEvaluator.EvaluateRDate() method is missing from this version. Why?

**Comment by rianjs** [comment]
I happen to remember that I deleted this method because it was a no-op method that instantiated a bunch of evaluators, but didn't actually evaluate anything. But I won't always remember why I delete one thing or another.

Are you having a problem where you think the solution is bringing back that particular method? If so, can you provide a test case so I can see it?

**Comment by rianjs** [comment]
Housekeeping: if you have a more specific question or bug, please re-open.

**Comment by robertsuditu** [comment]
Well, that method was very useful. See, when you have (e.g. daily) recurrent events and you want to move only one occurrence (e.g. The event "Do your cardio" happens daily at 8:00 am, but on one specific day you want to have it starting at 10:00 am), then you add an exception for that specific day starting at 8:00 am paired with an RDATE starting at 10:00 am. Without that method called, the GetOccurrences method would bring the list of occurrences all starting at 8:00 am, which would not be correct. The RDATE's must also be evaluated.

**Comment by rianjs** [comment]
Can you provide a repro VEVENT that demonstrates a bug?

**Comment by robertsuditu** [comment]
BEGIN:VCALENDAR
PRODID:-//ddaysoftware.com//NONSGML DDay.iCal 1.0//EN
VERSION:2.0
BEGIN:VEVENT
DTEND:20160829T090000
DTSTAMP:20160829T115057Z
DTSTART:20160829T080000
EXDATE:20160830T080000/PT1H
RDATE:20160830T100000/PT1H
RRULE:FREQ=DAILY
SEQUENCE:0
SUMMARY:Do your cardio
UID:abab717c-1786-4efc-87dd-6859c2b48eb6
END:VEVENT
END:VCALENDAR

**Comment by rianjs** [comment]
OK, so you've got a recurring event, whose first occurrence is Aug 29 at 8am. This event occurs daily.
- So we have a base set of occurrences at 8am.
- EXDATE deletes Aug 30[1]
- RDATE adds an occurrence on Aug 30 for 10am

So if we searched for a week's worth of occurrences starting on 8/29, we would expect to see a uniform set of events, _except for Aug 30 which would be at 10am_ instead of 8am.

Instead, we have uniform occurrences at 9am.

Repro:

``` csharp
public void RDateShouldBeUnionedWithRecurrenceSet()
{
    const string ical =
@"BEGIN:VCALENDAR
PRODID:-//ddaysoftware.com//NONSGML DDay.iCal 1.0//EN
VERSION:2.0
BEGIN:VEVENT
DTSTART:20160829T080000
DTEND:20160829T090000
EXDATE:20160830T080000/PT1H
RDATE:20160830T100000/PT1H
RRULE:FREQ=DAILY
SEQUENCE:0
SUMMARY:Do your cardio
UID:abab717c-1786-4efc-87dd-6859c2b48eb6
END:VEVENT
END:VCALENDAR";

    var collection = Calendar.LoadFromStream(new StringReader(ical));
    var firstEvent = collection.First().Events.First();
    var startSearch = new CalDateTime(DateTime.Parse("2015-08-28T07:00:00"), _tzid);
    var endSearch = new CalDateTime(DateTime.Parse("2016-08-28T07:00:00").AddDays(7), _tzid);

    var occurrences = firstEvent.GetOccurrences(startSearch, endSearch)
        .Select(o => o.Period as Period)
        .OrderBy(p => p.StartTime)
        .ToList();

    var expectedRDate = new CalDateTime(DateTime.Parse("2016-08-30T10:00:00"), _tzid);
    Assert.IsTrue(occurrences[1].StartTime.Equals(expectedRDate));
}
```

Worse, the first occurrence (which is supposed to be on Aug 29 at 8am) isn't found at all. That's a separate bug, and I'll file a separate ticket for it. Ugh.

Working hypothesis is the `EvaluateRDate` method I deleted a few months ago because "it didn't do anything".

**Comment by rianjs** [comment]
Here's the commit: https://github.com/rianjs/ical.net/commit/08a6802eda70d48842851cf80200363b70ff16da

Let's see if adding it back fixes the first bug.

**Comment by rianjs** [comment]
Fixed in f895fdc7c9dc2dcd9bfefd20dffba1824c550ea8 / Published in 2.2.12 available on nuget
